### PR TITLE
Added Delete endpoint

### PR DESCRIPTION
**Pull request by johnsn27**
Trello Ticket: https://trello.com/c/Fz363Cmt/41-delete-endpoint-for-unwatch-room

 ## Description

 ...

 ## Coding checklist

- [x] Trello ticket added.
- [x] Pull request URL in WhatsApp.

 ##### Review

- [ ] Two separate approvals.

### CODE REVIEW

File: restApi.py [hunk 1]
```diff
@@ -44,5 +44,16 @@ def get_watched_rooms():
    rows = c.fetchall()
    return json.dumps(results)    
 
+@app.route('/delete-watch/<int:watch>')
```
Comment (LanaCerise) [inline restApi.py]: Could you add `methods=['DELETE']` to this please?

File: restApi.py [hunk 2]
```diff
@@ -44,5 +44,16 @@ def get_watched_rooms():
...
+def delete_watched_rooms(watch):
+   conn = sqlite3.connect("BOOKING.db")
+   conn.row_factory = sqlite3.Row
+   results = []
+   
+   c = conn.cursor()
+   c.execute("DELETE FROM WATCHED WHERE WatchedId=%s" % watch)
+   conn.commit()
+   conn.close()
+
```
Comment (LanaCerise) [inline restApi.py:60]: I think we need a return statement for each endpoint.
After a quick test I'm getting:
``` ERROR in app: Exception on /delete-watch/19 [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1816, in full_dispatch_request
    return self.finalize_request(rv)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1831, in finalize_request
    response = self.make_response(rv)
  File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1957, in make_response
    'The view function did not return a valid response. The'
TypeError: The view function did not return a valid response. The function either returned None or ended without a return statement.
```
This did in fact delete the record from the database but it probably doesn't look good for us to have a 500 response.
Maybe the return value could be something like success or failure depending on whether the database call succeeded.

**Review by LanaCerise — CHANGES_REQUESTED** [review]

File: restApi.py [hunk 3]
```diff
@@ -44,5 +43,20 @@ def get_watched_rooms():
...
 
+@app.route('/delete-watch/<int:WatchId>', methods = ['DELETE', 'OPTIONS'])
+def delete_watched_rooms(WatchId):
+   conn = sqlite3.connect("BOOKING.db")
+   conn.row_factory = sqlite3.Row
+   results = []
+   
+   c = conn.cursor()
+   try:
+      c.execute("DELETE FROM WATCHED WHERE WatchedId=%s" % watch)
```
Comment (LanaCerise) [inline restApi.py]: 'watch' referenced here should be 'WatchId'

File: restApi.py [hunk 4]
```diff
@@ -44,5 +43,20 @@ def get_watched_rooms():
...
    return json.dumps(results)    
 
+@app.route('/delete-watch/<int:WatchId>', methods = ['DELETE', 'OPTIONS'])
+def delete_watched_rooms(WatchId):
+   conn = sqlite3.connect("BOOKING.db")
+   conn.row_factory = sqlite3.Row
+   results = []
+   
+   c = conn.cursor()
+   try:
```
Comment (LanaCerise) [inline restApi.py:53]: The try is currently still passing for me even when I try to use a WatchId that does not currently exist in the database.
Looks like we will probably need to use SQL to search for the WatchId first, as you previously suggested.

I'm not sure if this is too important though, as the UI shouldn't be giving users the option to unwatch rooms that are not on their watchlist.

File: restApi.py [hunk 5]
```diff
@@ -1,7 +1,6 @@
 import sqlite3
-from flask import Flask, request, jsonify
+from flask import Flask, request, jsonify, redirect, url_for
```
Comment (LanaCerise) [inline restApi.py]: `redirect` and `url_for` are unused so can be removed

File: restApi.py (hunk 4 above)
Comment (johnsn27) [reply restApi.py:53]: I think this is satisfied in the front end

**Pull request merged**